### PR TITLE
Updated uProf spack recipe to work with new download method.

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
We would like to take over the maintenance of this package going forward. Thank you, Carlos Bederián(zzzoom), for the initial setup.

### CODE REVIEW

File: var/spack/repos/builtin/packages/amduprof/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/amduprof/package.py:23]: Why are you taking over, instead of just adding a maintainer? Are you fine with that @zzzoom ?

File: var/spack/repos/builtin/packages/amduprof/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/amduprof/package.py:27]: Can you explain why do you need this dependency?

File: var/spack/repos/builtin/packages/amduprof/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/amduprof/package.py]: Can we use just ASCII as it was before?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/amduprof/package.py
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/amduprof/package.py:27]: CPU profiler need binutils for its disassembler. The disassembler can translate the binary code that the program is executing back into assembly language, which can then be analyzed by the profiler to understand how the program is using the CPU.
 Binutils is a collection of binary tools, including an assembler, linker, and binary utilities. It also includes a disassembler, which is why a CPU profiler needs it.

File: var/spack/repos/builtin/packages/amduprof/package.py
Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/amduprof/package.py]: Done.

File: var/spack/repos/builtin/packages/amduprof/package.py
Comment (zzzoom) [reply var/spack/repos/builtin/packages/amduprof/package.py:23]: Yeah, it's theirs after all.

**Comment by zzzoom** [comment]
LGTM.

Requiring a manual download is extremely annoying though, there are better ways to deal with EULAs.

File: var/spack/repos/builtin/packages/amduprof/package.py
Comment (tgamblin) [inline var/spack/repos/builtin/packages/amduprof/package.py:21]: Is there a direct download URL we can use here?  It would be way more convenient for users if they did not have to download the tarball manually.  

Do you need a spack-native way of having the user accept the EULA (e.g., like the way Debian does for some packages, where the user can run a command to indicate they accept?)

Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/amduprof/package.py:21]: Unfortunately direct uProf downloads without explicit EULA acceptance are not something that we are able to provide at this time.

It is possible that a Spack-native license acceptance mechanism could be a workable alternative, but would definitely require discussion with and careful review by our legal team.   We're completely open to exploring anything that would allow a better user experience, but we are very much constrained by our legal obligations.

Comment (tgamblin) [reply var/spack/repos/builtin/packages/amduprof/package.py:21]: ok let's keep it manual for now, but we can work on getting explicit license acceptance into spack.

**Review by tgamblin — APPROVED** [review]

**Comment by tgamblin** [comment]
LGTM. leaving it to @alalazo to resolve and merge.

**Review by alalazo — APPROVED** [review]

**Pull request merged**
